bundle/edu.gemini.spModel.core/src/test/scala/edu/gemini/spModel/core/Arbitraries.scala

```diff
@@ -50,6 +50,16 @@ trait Arbitraries {
       } yield Coordinates(ra, dec)
     }
 
+  // A way of picking coordinates within an offset distance in arcseconds from a given coordinate.
+  def genCoordsWithinDistance(c: Coordinates, lower: Double, upper: Double): Gen[Coordinates] =
+    for {
+      dRaSgn <- oneOf(-1.0, 1.0)
+      dDecSgn <- oneOf(-1.0, 1.0)
+      dRa <- choose(lower, upper).map(a => Angle.fromArcsecs(dRaSgn * a))
+      dDec <- choose(lower, upper).map(a => Angle.fromArcsecs(dDecSgn * a))
+    } yield c.offset(dRa, dDec)
```

Comment:
Thinking about it more, it seems like what we really want is an angular separation from `c` along a particular direction angle.  It seems like that will be more likely to land us closer to the border.  I found the math for this and typed it up, but I haven't tested it.  Maybe we should add this to `Coordinates` and test it.

```scala
  /**
   * Offset given angular distance and direction from these coordinates
   *
   * @param bearing angle clockwise from north
   * @param distance angular separation
   *
   * Source: http://www.movable-type.co.uk/scripts/latlong.html
   *
   * @return Coordinates at the given offset
   */
  def angularOffset(bearing: Angle, distance: Angle): Angle = {
    val  δ = distantance.toRadians
    val  θ = bearing.toRadians
    val φ1 = this.dec.toAngle.toRadians
    val λ1 = this.ra.toAngle.toRadians

    val φ2 = asin(sin(φ1) * cos(δ) + cos(φ1) * sin(δ) * cos(θ))
    val λ2 = λ1 + atan2(sin(θ) * sin(δ) * cos(φ1), cos(δ) - sin(φ1) * sin(φ2))

    val ra  = RightAscension.fromAngle(Angle.fromRadians(λ2);

    // Make sure the declination is in range.
    val dec = {
      val a0 = Angle.fromRadians(φ2).toDegrees
      Declination.fromAngle(a0).orElse {
        val a1 = if (a0 > 180) 540 - a0
                 else          180 - a0
        Declination.fromAngle(Angle.fromDegrees(a1))
      }
    }.get

    Coordinates(ra, dec)
  }
```

Reply:
Added to `Coordinates` and it seems to work properly with `Coordinates.angularDistance`.